pkg/prometheus/server/operator.go

```diff
@@ -1435,3 +1446,14 @@ func makeSelectorLabels(name string) map[string]string {
 		"prometheus":                    name,
 	}
 }
+
+// isOwnershipWrong returns false if the controller is the "owner" of the prometheus
+// or ownership has not been setted yet
+// Whether it's owner is determined by the value of label 'controllerID'
+// annotation. If the value matches the controllerID then it owns it.
+func (c *Operator) isOwnershipWrong(sset *appsv1.StatefulSet) bool {
+	if owner, ok := sset.ObjectMeta.Labels[prompkg.ControllerIDName]; ok {
```

Comment:
the design proposal (https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/proposals/202309-controller-id.md#how) suggested an annotation and I think that it's more appropriate than a label.

Reply:
https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/ 
`You can use either labels or annotations to attach metadata to Kubernetes objects. Labels can be used to select objects and to find collections of objects that satisfy certain conditions. In contrast, annotations are not used to identify and select objects.  `

Reply:
yes, if we're not going to need labels' features, an annotation is sufficient.

But we can go further, use a label, and add it as selector to all the lists and watches for Prometheus/AM etc., but this may imply more changes...


(maybe we should add a prefix to the annotation/label for more clarity)